addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -114,6 +114,20 @@ const GallerySliderWidget = publicWidget.Widget.extend({
         var self = this;
         this.$carousel = this.$el.is('.carousel') ? this.$el : this.$('.carousel');
         this.$indicator = this.$carousel.find('.carousel-indicators');
+        // restore BUTTON if they were removed with sanitize_form=True
+        if (this.$carousel.length && this.$carousel.find('button').length == 0) {
```

Comment:
- The first condition seems useless
- The second condition is too fragile:
    - if the carousel is well-formed but with `a` instead `button`, don't you break everything?
    - if the carousel content contains a button from another snippet / custom content, you don't fix it

Can't you instead rely on the presence of all the related bootstrap classes which are on those buttons?

Reply:
> this.$carousel[0].id

Oh, indeed

> I've adapted with class/attributes that I see in bootstrap 4 and 5 and on odoo 17 and 18 version.

I think you can base it on Bootstrap 5 only: the user should have immediately saw his carousel broken (+ we did not use buttons in Odoo 17.0 I think? That's why you target 18.0?). The only reason we do this is because we apparently cannot fix the "save" problem so we need new carousels to work in the future, but no need to restore carousel of old versions.